source/slang/slang-check-expr.cpp

```diff
@@ -2185,6 +2185,76 @@ namespace Slang
         return _canLValueCoerceScalarType(a, b);
     }
 
+
+    void SemanticsVisitor::compareMemoryQualifierOfParamToArgument(
+        ParamDecl* paramIn,
+        Expr* argIn)
+    {
+        auto arg = as<VarExpr>(argIn);
+        if (!paramIn || !arg)
+            return;
+
+        auto argDeclRef = arg->declRef;
+        if (!argDeclRef)
+            return;
+
+        ASTNodeType opThis;
+        ASTNodeType foundOp;
+        for (auto modThis : argDeclRef.getDecl()->modifiers)
+        {
+            UnownedStringSlice target;
+            opThis = modThis->astNodeType;
+            foundOp = ASTNodeType::NodeBase;
+            if (opThis == ASTNodeType::GloballyCoherentModifier)
+            {
+                foundOp = opThis;
+                target = UnownedStringSlice("coherent");
```

Comment:
You don't need this, just pass the modifier AST node to `diagnose` and it should work just fine. See `void printDiagnosticArg(StringBuilder& sb, Modifier* modifier)` defined in slang-ast-modifier.cpp.